File: cpp/RAM.cpp
#pragma once
#include "Komponen.cpp"
#include <iostream>
#include <string>
using namespace std;

class RAM : public Komponen
{
private:
    string tipe_RAM;
    int kapasitas;

public:
    RAM() {}

    RAM(string tipe_RAM, int kapasitas, string nama, string merk, int tahun_rilis, int harga) : Komponen(nama, merk, tahun_rilis, harga)
    {
        this->tipe_RAM = tipe_RAM;
        this->kapasitas = kapasitas;
    }

    void setTipeRAM(string tipe_RAM)
    {
        this->tipe_RAM = tipe_RAM;
    }

    void setKapasitas(int kapasitas)
    {
        this->kapasitas = kapasitas;
    }

    string getTipeRAM()
    {
        return tipe_RAM;
    }

    int getKapasitas()
    {
        return kapasitas;
    }

    ~RAM() {}
};

File: python/RAM.py
from Komponen import Komponen

class RAM(Komponen):
    def __init__(self, tipe_RAM="", kapasitas=0, nama="", merk="", tahun_rilis=0, harga=0):
        super().__init__(nama, merk, tahun_rilis, harga)
        self.tipe_RAM = tipe_RAM
        self.kapasitas = kapasitas
    
    def setTipeRAM(self, tipe_RAM):
        self.tipe_RAM = tipe_RAM
    
    def setKapasitas(self, kapasitas):
        self.kapasitas = kapasitas
    
    def getTipeRAM(self):
        return self.tipe_RAM
    
    def getKapasitas(self):
        return self.kapasitas